zilliqa/src/consensus.rs

```diff
@@ -564,58 +557,7 @@ impl Consensus {
                 warn!("state root hash prior to block execution mismatch, expected: {:?}, actual: {:?}.\nHead: {}", parent.state_root_hash(), self.state.root_hash()?, head_block);
             }
 
-            if !transactions.is_empty() {
-                trace!("applying {} transactions to state", transactions.len());
-            }
-
-            for txn in transactions {
-                let txn = txn.verify()?;
-                self.new_transaction(txn.clone())?;
-                let tx_hash = txn.hash;
-                if let Some(result) = self.apply_transaction(txn.clone(), parent.header)? {
-                    self.db
-                        .insert_block_hash_reverse_index(&tx_hash, &block.hash())?;
-                    let receipt = TransactionReceipt {
-                        block_hash: block.hash(),
-                        tx_hash,
-                        success: result.success,
-                        contract_address: result.contract_address,
-                        logs: result.logs,
-                        gas_used: result.gas_used,
-                    };
-                    info!(?receipt, "applied transaction {:?}", receipt);
-                    block_receipts.push(receipt);
-                } else {
-                    warn!("Failed to apply TX! Something might be wrong");
-                }
-            }
-
-            self.apply_rewards(&parent.committee, block.view(), &block.qc.cosigned)?;
-
-            // If we were the proposer we would've already processed the transactions
-            if !self.db.contains_transaction_receipts(&block.hash())? {
-                self.db
-                    .insert_transaction_receipts(&block.hash(), &block_receipts)?;
-            }
-
-            // Important - only add blocks we are going to execute because they can potentially
-            // overwrite the mapping of block height to block, which there should only be one of.
-            // for example, this HAS to be after the deal with fork call
-            self.add_block(block.clone())?;
-
-            if self.state.root_hash()? != block_state_root {
-                warn!(
-                    "State root hash mismatch! Our state hash: {}, block hash: {:?} block prop: {}",
-                    self.state.root_hash()?,
-                    block_state_root,
-                    block
-                );
-                return Err(anyhow!(
-                    "state root hash mismatch, expected: {:?}, actual: {:?}",
-                    block_state_root,
-                    self.state.root_hash()
-                ));
-            }
+            self.execute_block(&block, transactions)?;
```

Comment:
move the execution of a block into its own function because we potentially need to execute blocks when dealing with a fork